lua/ibl/init.lua

```diff
@@ -0,0 +1,381 @@
+        fn(bufnr)
+    end
+end
+
+---@param config ibl.config.full
+local setup = function(config)
+    M.initialized = true
+
+    if not config.enabled then
+        for bufnr, _ in pairs(global_buffer_state) do
+            clear_buffer(bufnr)
+        end
+        global_buffer_state = {}
+        inlay_hints.clear()
+        return
+    end
+
+    vim.schedule_wrap(function()
+        inlay_hints.setup()
+        highlights.setup()
+        autocmds.setup()
+
+        M.refresh_all()
+    end)()
+end
+
+--- Initializes and configures indent-blankline.
+---
+--- Optionally, the first parameter can be a configuration table.
+--- All values that are not passed in the table are set to the default value.
+--- List values get merged with the default list value.
+---
+--- `setup` is idempotent, meaning you can call it multiple times, and each call will reset indent-blankline.
+--- If you want to only update the current configuration, use `update()`.
+---@param config ibl.config?
+M.setup = function(config)
+    setup(conf.set_config(config))
+end
+
+--- Updates the indent-blankline configuration
+---
+--- The first parameter is a configuration table.
+--- All values that are not passed in the table are kept as they are.
+--- List values get merged with the current list value.
+---@param config ibl.config
+M.update = function(config)
+    setup(conf.update_config(config))
+end
+
+--- Overwrites the indent-blankline configuration
+---
+--- The first parameter is a configuration table.
+--- All values that are not passed in the table are kept as they are.
+--- All values that are passed overwrite existing and default values.
+---@param config ibl.config
+M.overwrite = function(config)
+    setup(conf.overwrite_config(config))
+end
+
+--- Configures indent-blankline for one buffer
+---
+--- All values that are not passed are cleared, and will fall back to the global config
+---@param bufnr number
+---@param config ibl.config
+M.setup_buffer = function(bufnr, config)
+    assert(M.initialized, "Tried to setup buffer without doing global setup")
+    bufnr = utils.get_bufnr(bufnr)
+    local c = conf.set_buffer_config(bufnr, config)
+
+    if c.enabled then
+        M.refresh(bufnr)
+    else
+        clear_buffer(bufnr)
+    end
+end
+
+--- Refreshes indent-blankline in all buffers
+M.refresh_all = function()
+    for _, win in ipairs(vim.api.nvim_list_wins()) do
+        vim.api.nvim_win_call(win, function()
+            M.refresh(vim.api.nvim_win_get_buf(win))
+        end)
+    end
+end
+
+local debounced_refresh = setmetatable({
+    timers = {},
+}, {
+    ---@param bufnr number
+    __call = function(self, bufnr)
+        bufnr = utils.get_bufnr(bufnr)
+        local config = conf.get_config(bufnr)
+        if not self.timers[bufnr] or self.timers[bufnr]:is_closing() then
+            ---@diagnostic disable-next-line:undefined-field
+            self.timers[bufnr] = vim.uv.new_timer()
```

Comment:
I'm getting an error that `vim.uv` is `nil`. Should it be `vim.loop`?
I'm using neovim v0.9.1.

Reply:
Yes, that is virtually the same thing. We could check nvim version (with features or the new lua api, although that wil ofcourse not work for older versions) wether we should use vim.uv or vim.loop

Reply:
vim.loop / vim.uv feature discussion https://github.com/neovim/neovim/pull/22846/

Reply:
Ah, my bad, vim.loop always exists. We should probably keep that then, for simplicity's sake. Even though the issue you mentioned specifies that we should probably not do that, but in the spirit of backwards compatibility I think we should as long as it doesn't become a mess.

Reply:
should be fixed now, thanks